composer/algorithms/agc/README.md

```diff
@@ -0,0 +1,97 @@
+# 📎 AGC
+
+[\[How to Use\]](#how-to-use) - [\[Suggested Hyperparameters\]](#suggested-hyperparameters) - [\[Technical Details\]](#technical-details) - [\[Attribution\]](#attribution)
+
+ `Computer Vision`
+
+AGC (Adaptive Gradient Clipping) .
+
+<!--| ![AGC](https://storage.googleapis.com/docs.mosaicml.com/images/methods/agc.png) |
+|:--:
+|*Need a picture.*|-->
+
+## How to Use
+
+### Functional Interface
+
+```python
+# Run the AGC algorithm directly on the model right after a loss.backward() call
+# using the Composer functional API.
+
+import torch
+import composer.functional as cf
+
+def training_loop(model, train_loader):
+    opt = torch.optim.Adam(model.parameters())
+    loss_fn = F.cross_entropy
+    model.train()
+  
+    for epoch in range(num_epochs):
+        for X, y in train_loader:
+            y_hat = model(X)
+            loss = loss_fn(y_hat, y)
+            loss.backward()
+            cf.apply_agc(model)
+            opt.step()
+            opt.zero_grad()
```

Comment:
minor, could you move the `opt.zero_grad()` to the top of the inner loop? I think that's how PyTorch recommends doing it these days, and also what we do in Composer

Reply:
done in 7b7bd14